include/termcolor/termcolor.hpp

```diff
@@ -39,7 +39,13 @@
 
 
 #include <iostream>
+#ifdef __STRICT_ANSI__  // Required when compiling with
```

Comment:
I did some testing and yes, you are right, it seems that the standard library in Cygwin is stricter than in other unix-like environments. And worst, this workaround won't work when _cstdio_ is included before _termcolor.h_.

A more correct approach when requiring strict standard conformance will be to `#undef __STRICT_ANSI__` whithin the _.cpp_ file that includes _termcolor.h_, above the inclusion of any header that may include _cstdio_, such as _iostream_, or above `#include <cstdio>` itself. That way we ensure the order of the `#include` directives won't have any impact on the compilation unit.

Please dismiss this request, I'll patch and send you a new one.

Results of my tests:

**g++ 4.9, 5.2, 5.3, and clang++ 3.7 on Cygwin 2.4:**
````sh
$ g++ -std=c++11 -c -o test.o test.cpp       : **ERROR**
$ g++ -std=gnu++11 -c -o test.o test.cpp     : **OK**
$ clang++ -std=c++11 -c -o test.o test.cpp   : **ERROR**
$ clang++ -std=gnu++11 -c -o test.o test.cpp : **OK**
````

**g++ 4.8 and clang++ 3.4 on FreeBSD 10.1:**

````sh
$ g++ -std=c++11 -c -o test.o test.cpp       : **OK**
$ g++ -std=gnu++11 -c -o test.o test.cpp     : **OK**
$ clang++ -std=c++11 -c -o test.o test.cpp   : **OK**
$ clang++ -std=gnu++11 -c -o test.o test.cpp : **OK**
````

**g++ 4.8 on Ubuntu 14.04:** (thanks to [jmaidana](https://github.com/jmaidana18))

````sh
$ g++ -std=c++11 -c -o test.o test.cpp       : **OK**
$ g++ -std=gnu++11 -c -o test.o test.cpp     : **OK**
````

In all cases the error was:

````sh
In file included from test.cpp:13:0:
termcolor.hpp: In function 'bool termcolor::__internal::is_atty(const ostream&)':
termcolor.hpp:449:46: error: 'fileno' was not declared in this scope
             return ::isatty(fileno(std_stream));
                                              ^
````